src/charm.py

```diff
@@ -58,11 +78,45 @@ def _on_peer_relation_changed(self, event):
             ] = event.relation.data[event.app]["MYSQL_ROOT_PASSWORD"]
             logger.info("Storing MYSQL_ROOT_PASSWORD in StoredState")
 
-    def _on_config_changed(self, _):
-        """This method handles the .on.config_changed() event"""
-        self._configure_pod()
+    def _on_config_changed(self, event):
+        """Set a new Juju pod specification"""
+        self.unit.status = MaintenanceStatus("Setting up containers.")
+        needs_restart = False
+        container = self.unit.get_container(PEER)
+        services = container.get_plan().to_dict().get("services", {})
```

Comment:
`services` always exists, so you can safely do this:
`service = container.get_plan().services.get(SERVICE)`
In your code `if not services` works just as fine because an empty dictionary evaluates as `False`.
If pebble devs decides to add some metadata to the service dictionary, this code will be incorrect.

Reply:
AFAIU `services` not always exists, for instance:

`container.get_plan().services.get("mysql")` returns None

Reply:
this `None` is from `get("mysql")`, not from `services`.

Reply:
Following in #45